pkg/action/install.go

```diff
@@ -71,7 +71,7 @@ type Install struct {
 	ClientOnly               bool
 	Force                    bool
 	CreateNamespace          bool
-	DryRun                   bool
+	DryRun                   string
```

Comment:
This is a breaking api change. `DryRun` needs to remain a bool. Perhaps add a new member, `DryRunOption`. All the cli flags can work using the new member but the old member would still need to be honored if true as there are other applications out there that use this as a library.

Reply:
And what do you think of that ?
https://github.com/helm/helm/issues/8137#issuecomment-1383664380

Reply:
Understood. Let me rework it with another flag.

Reply:
Shoot, not quite what I was trying to suggest. I was still suggesting to just have the one single `dryrun` flag for the cli. That flag would load its input into the new `DryRunOption` string.

The reason for the new string values is that if a downstream application is using the install action as a library, it's going to have an `Install` struct with a `DryRun` boolean already. If we change the type of `DryRun`, those downstream applications will break.

So my suggestion is that the new string element would be used for the dryrun flag. In places where DryRun is checked, call a function like:
```go
func (i *Install) DryRunCheck() string {
	if i.DryRunOptions != nil {
		return i.DryRunOptions
	}
	if i.DryRun {
		return "client"
	}
	return "none"
}
```
you could simplify all the choices here and use/return a custom type instead of string, which might make the rest of the code easier:
```
type DryRunType string

var DryRunServer = DryRunType("server")
var DryRunClient = DryRunType("client")
var DryRunNone = DryRunType("none")
```

```
if DryRunCheck() == DryRunServer { ...
```

The benefit to all this is that the user experience at the command line doesn't have to change, yet we maintain backward compatibility for all the projects that use helm as a go library.

disclaimer: These snippets are used as illustrations. Don't feel like I'm dictating anything.

Reply:
Don't you love it when you reply to something before you've scrolled all the way down?